runtime/compiler/runtime/J9CodeCacheManager.cpp

```diff
@@ -566,40 +558,40 @@ J9::CodeCacheManager::chooseCacheStartAddress(size_t repositorySize)
 #if defined(TR_HOST_64BIT) && defined(TR_TARGET_X86)
    if (!TR::Options::getCmdLineOptions()->getOption(TR_DisableSmartPlacementOfCodeCaches))
       {
-      size_t alignment = 2 * 1024 * 1024; // 2MB alignment
+      size_t alignment = 2 * 1024 * 1024; // 2MB alignment is preferred on x86-64
       TR::CodeCacheConfig & config = self()->codeCacheConfig();
       const size_t largeCodePageSize = config.largeCodePageSize();
 
       // at least on Windows we need 2MB minimum alignment for 64bit
       if (largeCodePageSize > alignment)
          alignment = largeCodePageSize;
 
-      size_t safeDistance = repositorySize + SAFE_DISTANCE_REPOSITORY_JITLIBRARY;
       uintptr_t someFunctionPointer = self()->getSomeJitLibraryAddress();
 
       mcc_printf("addCodeCache() function address is %p\n", someFunctionPointer);
 
-      // if we are using 1GB pages or something even larger
-      if (largeCodePageSize >= 0x40000000)
+      // Depending on where the JIT dll is placed in memory, we may want
+      // to allocate the code cache before or after the JIT dll
+      if (someFunctionPointer > MAX_DISTANCE_NEAR_JITLIBRARY_TO_AVOID_TRAMPOLINE)
          {
-         if (someFunctionPointer > largeCodePageSize * 2)
+         // Enough space is available before the JIT dll
+         // For Linux we can afford to search a larger space (~2 GB) because we use smaps to accelerate the search operation.
+         // Also, if we use very large pages (1 GB), then we can afford to search a larger space because there won't be
+         // too many attempts to find the right spot.
+         if (TR::Compiler->target.isLinux() || alignment > 2 * 1024 * 1024)
```

Comment:
Did you mean this to be 
```
|| alignment >= 1 * 1024 * 1024 * 1024
```
? Or is it that on windows, the only alignment possible after 2MB is 1GB?

Reply:
For 2MB pages we can search a limited address range like in the original implementation.
For 1GB pages we can search the entire 2 GB address range since we jump page by page.